MarkBind/markbind
# Support favicon

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Documentation update
• [X] New feature



Fixes #96.



**What is the rationale for this request?**
User would like to add a favicon to their website. This is only possible if the HTML `<head>` tag contains a link to the image of the favicon.

**What changes did you make? (Give an overview)**
* Modified `page.ejs` to insert the favicon path if it is enabled.
* Added the `favIconPath` parameter in `site.json` and passed it to the page template generator.
* Modified the user guide to mention about the new parameter.

**Provide some example code that this change will affect:**


After:
```html
<html>
  <head>
    ...
    <link rel="icon" href="favicon.png"> 
    ...
  </head>
   ...
</html>
```

**Is there anything you'd like reviewers to focus on?**
The documentation & tests.

**Testing instructions:**
1. Insert an additional line to a markbind website's `site.json`:
```json
"favIconPath": "favicon.png",
```
1. Add a `favicon.png` to the root directory of the markbind website.
1. Execute `markbind serve`.
1. The new webpage should now show the favicon on the browser's title bar.
